sdcm/cluster.py

```diff
@@ -2393,21 +2394,46 @@ def run_nodetool(self, sub_cmd, args="", options="", timeout=None,
                 self.generate_coredump_file()
             raise
 
-    def check_node_health(self):
+    def check_node_health(self, retries: int = CHECK_NODE_HEALTH_RETRIES) -> None:
         # Task 1443: ClusterHealthCheck is bottle neck in scale test and create a lot of noise in 5000 tables test.
         # Disable it
         if not self.parent_cluster.params.get('cluster_health_check'):
             return
 
-        nodes_status = self.get_nodes_status()
-        peers_details = self.get_peers_info() or {}
-        gossip_info = self.get_gossip_info() or {}
+        # Make several attempts to validate node health.
+        # All attempts except last will break after the first issue to retry again.
+        for publish in (False, ) * (retries - 1) + (True, ):
```

Comment:
too complex, better to simlify to:
```python
error_events = []
for _ in range(retries):
    ...
    error_events = list(itertools.chain(...))
    if error_events:
        time.sleep(CHECK_NODE_HEALTH_RETRY_DELAY)
        continue
    else:
           break

    
if error_events:
    for event in events:
        event.publish()


```

Reply:
This will break the idea to stop on the first validation error.

Reply:
simplified